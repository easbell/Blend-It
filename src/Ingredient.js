import React, {Component} from 'react';
import IngredientContainer from './IngredientContainer.js';
import './Ingredient.css';

export default class Ingredient extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isChosen: false
    }
  }

  selectIngredient = () => {
    const {isChosen} = this.state
    const {ingredient, chooseIngredients} = this.props
    this.setState({isChosen: !isChosen}, chooseIngredients(ingredient))
    
  }

  removeIngredient = () => {
    const {isChosen} = this.state
    const {ingredient, removeIngredient} = this.props
    this.setState({isChosen: !isChosen}, removeIngredient(ingredient))
  }

  render() {
    const {isChosen} = this.state
    const {ingredient} = this.props
    return ( 
      <div>
        {!isChosen && 
          <h4 className='not-selected' onClick={this.selectIngredient}>
            {ingredient}
          </h4>
        }
        {isChosen && 
          <h4 className='selected' onClick={this.removeIngredient}>
            {ingredient}
          </h4>
        }
      </div>
    )
  }
}
