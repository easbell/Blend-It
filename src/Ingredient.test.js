import React from 'react';
import Ingredient from './Ingredient.js';
import { shallow } from 'enzyme';

const mockIngredient = 'Banana';
const chooseIngredientsMock = jest.fn();
const removeIngredientMock = jest.fn();

describe('Ingredient', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Ingredient 
        ingredient={mockIngredient} 
        key={mockIngredient} 
        chooseIngredients={chooseIngredientsMock}
        removeIngredient={removeIngredientMock}
      />
    )
  });

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have proper default states', () => {
    expect(wrapper.state()).toEqual({isChosen: false})
  });

  it('should change state and call proper function when selectIngredient is called', () => {
    expect(wrapper.state()).toEqual({isChosen: false});
    wrapper.find('.not-selected').simulate('click');
    expect(wrapper.state()).toEqual({isChosen: true});
    expect(chooseIngredientsMock).toBeCalled();
  });

  // it('should toggle state back and call proper function when selectIngredient is called again', () => {
  //   expect(wrapper.state()).toEqual({isChosen: true});
  //   wrapper.find('.selected').simulate('click');
  //   expect(wrapper.state()).toEqual({isChosen: false});
  //   expect(removeIngredientMock).toBeCalled();
  // });
})