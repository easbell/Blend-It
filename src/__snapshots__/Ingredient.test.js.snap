// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredient should match a snapshot with all data passed in 1`] = `
<div
  className="ingredient-list"
>
  <h4
    className="not-selected"
    onClick={[Function]}
  >
    Banana
  </h4>
</div>
`;
